test/e2e/fail_forward_e2e_test.go

```diff
@@ -119,36 +214,41 @@ var _ = Describe("Fail Forward Upgrades", func() {
 			failedInstallPlanRef = subscription.Status.InstallPlanRef
 			_, err = fetchInstallPlan(GinkgoT(), crclient, failedInstallPlanRef.Name, failedInstallPlanRef.Namespace, buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseFailed))
 			Expect(err).To(BeNil())
-
-			By("updating the operator group remove service account without permissions")
-			Eventually(operatorGroupServiceAccountNameSetter(crclient, generatedNamespace.GetName(), ogName, "")).Should(Succeed())
 		})
 		AfterEach(func() {
 			By("removing the testing catalog resources")
-			Expect(magicCatalog.UndeployCatalog(context.Background())).To(BeNil())
+			for _, cleanup := range cleanups {
+				cleanup()
+			}
 		})
 		It("eventually reports a successful state when multiple bad versions are rolled forward", func() {
-			By("patching the OperatorGroup to reduce the bundle unpacking timeout")
-			addBundleUnpackTimeoutOGAnnotation(context.Background(), c, types.NamespacedName{Name: ogName, Namespace: generatedNamespace.GetName()}, "1s")
-
-			By("patching the catalog with a bad bundle version")
-			badProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fail-forward/multiple-bad-versions", "example-operator.v0.2.1-non-existent-tag.yaml"))
-			Expect(err).To(BeNil())
-			err = magicCatalog.UpdateCatalog(context.Background(), badProvider)
-			Expect(err).To(BeNil())
+			By("updating the catalog with a v0.2.1 bundle that has an invalid CSV")
+			cleanup, deployError := updateCatalogSource(generatedNamespace.GetName(), catalogSourceName, "v0.1.0", "v0.2.0-invalid", "v0.2.1-invalid")
+			Expect(deployError).To(BeNil())
+			if cleanup != nil {
+				cleanups = append(cleanups, cleanup)
+			}
 
-			By("waiting for the subscription to maintain the example-operator.v0.2.0 status.currentCSV")
-			Consistently(subscriptionCurrentCSVGetter(crclient, subscription.GetNamespace(), subscription.GetName())).Should(Equal("example-operator.v0.2.0"))
```

Comment:
This was only valid when the bundle failed to unpack (and, anyway, in the distributed system is not a good way to prove something unless you wait forever :) )